src/task.c

```diff
@@ -168,24 +266,43 @@ typedef struct {
    platform_heap_id heap_id;
 } thread_invoke;
 
+/*
+ * -----------------------------------------------------------------------------
+ * task_invoke_with_hooks() - Single interface to invoke a user-specified
+ * call-back function, 'func', to perform Splinter work.
+ *
+ * A thread has been created with this function as the worker function. Also,
+ * the thread-creator has registered another call-back function to execute.
+ * This function invokes that call-back function bracketted by calls to
+ * register / deregister the thread with Splinter's task system. This allows
+ * the call-back function 'func' to now call Splinter APIs to do diff things.
+ * -----------------------------------------------------------------------------
+ */
 static void
 task_invoke_with_hooks(void *func_and_args)
 {
-   thread_invoke         *thread_to_start = (thread_invoke *)func_and_args;
-   platform_thread_worker func            = thread_to_start->func;
-   void                  *arg             = thread_to_start->arg;
+   thread_invoke         *thread_started = (thread_invoke *)func_and_args;
```

Comment:
Renamed variable from `thread_to_start` to `thread_started`, because by the time this function executes, it's already executing in the context of a newly fired up thread.

(Prior variable name was meaningful & correct in this fn's caller, where the thread was just going to be started.)